From da1710bbfbdd948688357ce8cd7af656a17680bd Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Fri, 8 Nov 2024 08:25:19 +0100
Subject: [PATCH] mtd: hyperbus: hyperbus-stm32: set pin muxing configuration
 on suspend/resume

Set pin muxing configuration on suspend/resume.

Change-Id: I24ba6c0a141365887d5092c3ffd6d7f78cbb18e8
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/417262
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
---
 drivers/mtd/hyperbus/hyperbus-stm32.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/mtd/hyperbus/hyperbus-stm32.c
+++ b/drivers/mtd/hyperbus/hyperbus-stm32.c
@@ -250,7 +250,7 @@ static int __maybe_unused stm32_hyperbus
 
 	pm_runtime_put_sync_suspend(omi->dev);
 
-	pinctrl_pm_select_sleep_state(dev);
+	pinctrl_pm_select_sleep_state(omi->dev);
 
 	return pm_runtime_force_suspend(omi->dev);
 }
@@ -265,7 +265,7 @@ static int __maybe_unused stm32_hyperbus
 	if (ret < 0)
 		return ret;
 
-	pinctrl_pm_select_default_state(dev);
+	pinctrl_pm_select_default_state(omi->dev);
 
 	ret = pm_runtime_resume_and_get(omi->dev);
 	if (ret < 0)
